src/main/java/fi/vrk/xroad/monitor/actor/Supervisor.java

```diff
@@ -0,0 +1,98 @@
+package fi.vrk.xroad.monitor.actor;
+
+import akka.actor.*;
+import akka.dispatch.Futures;
+import akka.japi.pf.DeciderBuilder;
+import akka.pattern.Patterns;
+import akka.routing.SmallestMailboxPool;
+import akka.util.Timeout;
+import fi.vrk.xroad.monitor.extensions.SpringExtension;
+import fi.vrk.xroad.monitor.parser.SecurityServerInfo;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+import scala.concurrent.Await;
+import scala.concurrent.Future;
+import scala.concurrent.duration.Duration;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static akka.actor.SupervisorStrategy.*;
+import static fi.vrk.xroad.monitor.util.MonitorCollectorConstants.SUPERVISOR_MONITOR_DATA_ACTOR_POOL_SIZE;
+
+/**
+ * Supervisor for actors
+ */
+@Component
+@Scope("prototype")
+@Slf4j
+public class Supervisor extends AbstractActor {
```

Comment:
I recommend testing basic error handling (and whether the supervisor strategies work as expected) by throwing some exceptions from non-supervisor actors.